src/sentry/migrations/0294_backfill_alert_owners.py

```diff
@@ -0,0 +1,63 @@
+# Generated by Django 2.2.27 on 2022-04-08 22:24
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def backfill_alert_owners(apps, schema_editor):
+    AlertRule = apps.get_model("sentry", "AlertRule")
+    User = apps.get_model("sentry", "User")
+    OrganizationMember = apps.get_model("sentry", "OrganizationMember")
+
+    for alert_rule in RangeQuerySetWrapperWithProgressBar(AlertRule.objects_with_snapshots.all()):
```

Comment:
Does `objects_with_snapshots` work here? I'd assume it doesn't, since we usually don't have access to the real models in migrations.

Reply:
Looks like I'm wrong here - I'm writing a similar migration and it does work.

Reply:
I think it should work - see [Gilbert's previous migration](https://github.com/getsentry/sentry/blob/1b8c280e5206e40c0b63c5537e695f11baab5d87/tests/sentry/migrations/test_0286_backfill_alertrule_organization.py#L16).

Reply:
In the end I found out that that this does cause problems. I was getting random failures in this pr about my column not existing: https://github.com/getsentry/sentry/pull/36292

Turns out that since we access this via `objects_with_snapshots` it uses the real model, rather than the one that is incrementally built up in the migration framework. This then causes those migrations to fail. So ideally we should just use `objects` here, it won't make any difference to what you're doing either way.

Reply:
I'm seeing test failures with `AttributeError: type object 'AlertRule' has no attribute 'objects'` if I switch to `AlertRule.objects`. Any ideas on how to fix that?

Reply:
There's something weird going on here... Stick with `objects_with_snapshots` in this pr and I'll work out what's wrong in mine